samples/sample-app-storage/src/main/java/com/azure/android/storage/sample/MsalClient.java

```diff
@@ -0,0 +1,94 @@
+package com.azure.android.storage.sample;
+
+import android.app.Activity;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.azure.android.storage.sample.core.credential.AccessToken;
+import com.azure.android.storage.sample.core.util.tokenrequest.TokenResponseCallback;
+import com.microsoft.identity.client.AuthenticationCallback;
+import com.microsoft.identity.client.IAuthenticationResult;
+import com.microsoft.identity.client.PublicClientApplication;
+import com.microsoft.identity.client.exception.MsalClientException;
+import com.microsoft.identity.client.exception.MsalException;
+import com.microsoft.identity.client.exception.MsalServiceException;
+import com.microsoft.identity.client.exception.MsalUiRequiredException;
+
+import org.threeten.bp.DateTimeUtils;
+import org.threeten.bp.OffsetDateTime;
+import org.threeten.bp.ZoneOffset;
+
+import java.util.Date;
+
+class MsalClient {
+    private static final String TAG = MsalClient.class.getSimpleName();
+
+    private MsalClient() {}
+
+    public static void signIn(@NonNull final PublicClientApplication aadApp,
+                       @NonNull final Activity activity,
+                       @NonNull final String[] scopes,
+                       @NonNull final TokenResponseCallback callback) {
+        aadApp.getAccounts(accounts -> {
+            if (!accounts.isEmpty()) {
+                aadApp.acquireTokenSilentAsync(scopes, accounts.get(0), new AuthenticationCallback() {
+                    @Override
+                    public void onSuccess(IAuthenticationResult authResult) {
+                        Log.d(TAG, "[acquireTokenSilentAsync]: Successfully authenticated.");
+                        callback.onToken(new AccessToken(authResult.getAccessToken(),
+                                toOffsetDateTime(authResult.getExpiresOn())));
+                    }
+
+                    @Override
+                    public void onError(MsalException exception) {
+                        Log.d(TAG, "[acquireTokenSilentAsync]: Authentication failed: " + exception.toString());
+                        if (exception instanceof MsalClientException) {
```

Comment:
Can we add a comment here about what this type of exception means?

Reply:
it's for general local error in msal library. 

@vcolin7 could you please add a comment referring to https://javadoc.io/doc/com.microsoft.identity.client/msal/0.1.1/com/microsoft/identity/client/MsalClientException.html

Reply:
Sure thing